src/main/target/OMNINXT/hardware_revision.c

```diff
@@ -0,0 +1,300 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "platform.h"
+
+#ifdef USE_HARDWARE_REVISION_DETECTION
+
+#include "build/debug.h"
+
+#include "drivers/adc_impl.h"
+#include "drivers/io_types.h"
+#include "drivers/io.h"
+#include "drivers/io_impl.h"
+#include "drivers/rcc.h"
+#include "drivers/time.h"
+
+#include "hardware_revision.h"
+
+#undef DEBUG_HARDWARE_REVISION_ADC
+#undef DEBUG_HARDWARE_REVISION_TABLE
+
+uint8_t hardwareRevision = 0;
+
+// Do ADC on IDDetectPin and determine revision
+// If VREFINT is used, we can (probably) get a pretty good precision
+// that we can distinguish tens of different voltages.
+
+#define ADC_ID_DETECT_PIN PC2
+
+typedef struct idDetect_s {
+    uint32_t ratio;
+    uint8_t revision;
+} idDetect_t;
+
+// To deploy the analog ID detection in production:
+// - Need some theoretical evaluation and experimentation to determine
+//   IDDET_ERROR value (ADC with VREFINT compensation is quite accurate).
+// - Do some planning on revision numbering scheme.
+// - Divider value planning for the scheme (separation).
+
+#define IDDET_RATIO(highside, lowside) ((lowside) * 1000 / ((lowside) + (highside)))
+#define IDDET_ERROR 12
+
+static idDetect_t idDetectTable[] = {
+#ifdef OMNINXT7
+    { IDDET_RATIO(10000, 10000), 1 },
```

Comment:
Aren't these lines behind conditional compilation identical?

Reply:
They are identical at the moment, but I'm expecting them to diverge as variants are produced based on NXT architecture.

Reply:
Can't wait to learn about these variants :-D
Are there going to be some with built-in PDB?